int16 percentage
bool dc_in
bool charging